### PR TITLE
fix(hoshi): avoiding unnecessary updates

### PR DESCRIPTION
**Pull request by Sh4rky**
I avoid unnecessary reconfiguration for each time the draw method was being called, also indented properly.
And changed the example to use a dynamic tableview

hoshitextfield

Screenshots for both standard sizes and colors
simulator screen shot - iphone xr - 2018-12-28 at 15 31 46
simulator screen shot - iphone xr - 2018-12-28 at 15 31 52

### CODE REVIEW

File: TextFieldEffects/TextFieldsDemo/ExampleTableViewController.swift [hunk 1]
```diff
@@ -19,26 +19,84 @@ class ExampleTableViewController : UITableViewController, UITextFieldDelegate {
    textFields prepopulated with the name "Raul" (for testing purposes)
    */
   let prefillTextFields = false
+  var cells = [Any](repeating: 0, count: 2)
 
   override func viewDidLoad() {
     super.viewDidLoad()
-    guard prefillTextFields == true else { return }
+//    guard prefillTextFields == true else { return }
```
Comment (dhvlshah88) [inline TextFieldEffects/TextFieldsDemo/ExampleTableViewController.swift]: I know its example but I don't think you need to keep commented code.

Comment (Sh4rky) [reply TextFieldEffects/TextFieldsDemo/ExampleTableViewController.swift]: you are absolutely right, I forgot about this

**Review by xdansolo32 — APPROVED** [review]
slick~~

File: TextFieldEffects/TextFieldsDemo/ExampleTableViewController.swift [hunk 2]
```diff
@@ -19,26 +19,80 @@ class ExampleTableViewController : UITableViewController, UITextFieldDelegate {
...
+
+    })
+    delete.backgroundColor = UIColor.red
+    return [delete]
+  }
+
+  override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+    return cells.count
   }
+  override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
```
Comment (andrewredko) [inline TextFieldEffects/TextFieldsDemo/ExampleTableViewController.swift:61]: suggestion: add space line between methods

**Review by andrewredko — COMMENTED** [review]
suggestion: for demo project, can you setup fields closer to TB app style, like red color for error state, different font weight/size for placeholder and value?

File: TextFieldEffects/TextFieldsDemo/ExampleTableViewController.swift [hunk 3]
```diff
@@ -19,26 +19,80 @@ class ExampleTableViewController : UITableViewController, UITextFieldDelegate {
...
+    guard let cell = tableView.dequeueReusableCell(withIdentifier: "MyAwesomeCell") else {
+      return UITableViewCell()
+    }
 
+    let textField = HoshiTextField(frame: cell.frame)
+    if indexPath.row == 0 {
+      textField.placeholderLabel.text = "Firstname"
```
Comment (prempratapsingh) [inline TextFieldEffects/TextFieldsDemo/ExampleTableViewController.swift]: Suggestion: Would be nice to use localisation for these texts.

File: TextFieldEffects/TextFieldsDemo/ExampleTableViewController.swift [hunk 4]
```diff
@@ -19,26 +19,80 @@ class ExampleTableViewController : UITableViewController, UITextFieldDelegate {
...
+      textField.placeholder = "Firstname"
+    } else {
+      textField.placeholderLabel.text = "Lastname"
```
Comment (prempratapsingh) [inline TextFieldEffects/TextFieldsDemo/ExampleTableViewController.swift]: Suggestion: Would be nice to use localisation for these texts.

File: TextFieldEffects/TextFieldsDemo/ExampleTableViewController.swift [hunk 5]
```diff
@@ -19,26 +19,80 @@ class ExampleTableViewController : UITableViewController, UITextFieldDelegate {
...
-    hoshiTextField?.hideError()
+    guard let textField = textField as? HoshiTextField else { return }
+    textField.hideError()
   }
   @available(iOS 10.0, *)
   func textFieldDidEndEditing(_ textField: UITextField, reason: UITextFieldDidEndEditingReason) {
-    if let text = textField.text, !text.isEmpty {
-      hoshiTextField?.showError(message: "Bad Name!")
+    if let text = textField.text, !text.isEmpty, let textField = textField as? HoshiTextField {
+      textField.showError(message: "Bad Name!")
```
Comment (prempratapsingh) [inline TextFieldEffects/TextFieldsDemo/ExampleTableViewController.swift]: Suggestion: Would be nice to use localisation for these texts.

**Review by prempratapsingh — COMMENTED** [review]
Looks good, just a few suggestions.

**Pull request merged**